src/main/java/org/datadog/jmxfetch/util/CustomLogger.java

```diff
@@ -1,123 +1,192 @@
+                );
+            }
+        };
+
+        // log level
+        Level julLevel = level.toJulLevel();
+
+        // prepare the different handlers
+        ConsoleHandler stdoutHandler = null;
+        ConsoleHandler stderrHandler = null;
+        FileHandler fileHandler = null;
+
+        // the logLocation isn't always containing a file, it is sometimes
+        // referring to a standard output. We want to create a FileHandler only
+        // if the logLocation is a file on the FS.
+        if (logLocation != null && logLocation.length() > 0) {
+            if (!isStdOut(logLocation) && !isStdErr(logLocation)) {
+                // file logging
+                try {
+                    // maximum one 5MB file
+                    fileHandler = new FileHandler(logLocation, MAX_FILE_SIZE, FILE_COUNT);
+                    fileHandler.setFormatter(formatter);
+                    // note: fileHandler defaults to Level.ALL, so no need to set its log level
+                } catch (Exception e) {
+                    fileHandler = null;
+                    log.error("can't open the file handler:", e);
+                }
+            } else if (isStdErr(logLocation)) {
+                // console handler sending on stderr
+                // note that ConsoleHandler is sending on System.err
+                stderrHandler = new ConsoleHandler();
+                stderrHandler.setFormatter(formatter);
+                stderrHandler.setLevel(julLevel);
             }
         }
 
-        // replace default appender with the correct layout
-        LoggerConfig loggerConfig = config.getLoggerConfig(LogManager.ROOT_LOGGER_NAME);
-        loggerConfig.removeAppender(target);
+        // always have a console handler sending the logs to stdout
+        stdoutHandler = new StdoutConsoleHandler();
+        stdoutHandler.setFormatter(formatter);
+        stdoutHandler.setLevel(julLevel);
 
-        Appender appender = ConsoleAppender.newBuilder()
-                    .setConfiguration(config)
-                    .withName(target)
-                    .withLayout(layout)
-                    .build();
-        appender.start();
+        logger.setLevel(julLevel);
 
-        loggerConfig.addAppender(appender, null, null);
-        loggerConfig.setLevel(level);
+        // set our configured handlers
+        if (fileHandler != null) {
+            logger.addHandler(fileHandler);
+        }
+        if (stdoutHandler != null) { // always non-null but doesn't cost much
+            logger.addHandler(stdoutHandler);
+        }
+        if (stderrHandler != null) {
+            logger.addHandler(stderrHandler);
+        }
+
+        // filter all other log handlers
+        for (Enumeration<String> logNames = manager.getLoggerNames() ;
+                logNames.hasMoreElements() ; ) {
+            String logName = logNames.nextElement();
+
+            for (Handler handler : manager.getLogger(logName).getHandlers()) {
```

Comment:
According to java docs, `manager.getLogger` can return `null` so we need to check for that value  https://docs.oracle.com/javase/7/docs/api/java/util/logging/LogManager.html#getLogger(java.lang.String)

Reply:
also, I think we might as well just remove the handlers we find instead of setting a filter on them?

Reply:
actually, I think all of this is unnecessary because `manager.reset();` above should already remove all of the handlers: https://docs.oracle.com/javase/7/docs/api/java/util/logging/LogManager.html#reset():

> For all named loggers, the reset operation removes and closes all Handlers and (except for the root logger) sets the level to null. The root logger's level is set to Level.INFO.

So, once `manager.reset()` is called, I think all existing handlers are removed.

Reply:
(openJDK source code, for reference: https://github.com/openjdk-mirror/jdk7u-jdk/blob/f4d80957e89a19a29bb9f9807d2a28351ed7f7df/src/share/classes/java/util/logging/LogManager.java#L832-L845 )